src/views/collection-view/components/collection-meta-progress/helpers/parse-collection-progress.ts

```diff
@@ -0,0 +1,171 @@
+	progressData: ApiCollectionTutorialProgress[],
+	tutorials: TutorialLite[],
+	collection: Pick<Collection, 'id' | 'slug'>
+) {
+	/**
+	 * The basics
+	 */
+	const tutorialCount = tutorials.length
+	const inProgressTutorialCount = countProgressedRecords(
+		progressData || [],
+		collection.id,
+		TutorialProgressStatus.in_progress
+	)
+	const completedTutorialCount = countProgressedRecords(
+		progressData || [],
+		collection.id,
+		TutorialProgressStatus.complete
+	)
+	const isCompleted = completedTutorialCount == tutorialCount
+	const isInProgress =
+		inProgressTutorialCount > 0 || (completedTutorialCount > 0 && !isCompleted)
+	/**
+	 * Tutorial CTA
+	 */
+	const tutorialCta = getNextTutorialCta({
+		progressData,
+		tutorials,
+		collectionSlug: collection.slug,
+		isCompleted,
+		isInProgress,
+		completedTutorialCount,
+		tutorialCount,
+	})
+	/**
+	 * Status label
+	 */
+	let statusLabel: string
+	if (isCompleted) {
+		statusLabel = 'Complete'
+	} else if (isInProgress) {
+		statusLabel = `${completedTutorialCount}/${tutorialCount}`
+	} else {
+		statusLabel = `${tutorialCount} tutorial${tutorialCount == 1 ? '' : 's'}`
+	}
+	/**
+	 * Return it all
+	 */
+	return {
+		completedTutorialCount,
+		isCompleted,
+		isInProgress,
+		statusLabel,
+		tutorialCount,
+		tutorialCta,
+	}
+}
+
+/**
+ * Given an array of progress records, and a collection.id,
+ * Return the count of records that are completed in that collection.
+ */
+function countProgressedRecords(
+	progressData: ApiCollectionTutorialProgress[],
+	collectionId: Collection['id'],
+	targetProgress: TutorialProgressStatus
+): number {
+	return progressData.filter((record: ApiCollectionTutorialProgress) => {
+		return (
+			record.collection_id == collectionId &&
+			progressPercentToStatus(record.complete_percent) == targetProgress
+		)
+	}).length
+}
+
+/**
+ * Given progress and tutorial data for particular collection,
+ * Return a CTA object that links to the "next tutorial" in the collection.
+ *
+ * For in-progress collections, the "next tutorial" is the first tutorial
+ * in the collection that does not have "complete" as its progress status.
+ *
+ * For completed collections, and for all unauthenticated cases,
+ * the "next tutorial" is the first tutorial in the collection.
+ */
+function getNextTutorialCta({
+	progressData,
+	tutorials,
+	isCompleted,
+	isInProgress,
+	collectionSlug,
+	completedTutorialCount,
+	tutorialCount,
+}: {
+	progressData: ApiCollectionTutorialProgress[]
+	tutorials: TutorialLite[]
+	isCompleted: boolean
+	isInProgress: boolean
+	collectionSlug: Collection['slug']
+	completedTutorialCount: number
+	tutorialCount: number
+}) {
+	/**
+	 * If we're not authenticated or have no progress,
+	 * we'll use the first tutorial.
+	 *
+	 * If we do have progress data, we return the first incomplete tutorial,
+	 * or if all tutorials are complete, then the first tutorial.
+	 */
+	let targetTutorial
+	if (isCompleted || !progressData) {
```

Comment:
non-blocking optional suggestion: Pulling the logic out to compute the `targetTutorial` might make this function easier to reason about.

Reply:
Sweet! I'm going to refine this further in #1046 👍